### PR TITLE
DCGAN's Discriminator : add `nn.BatchNorm2d(ndf)`  to fix #1126

### LINKED ISSUE #1126: DCGAN's Discriminator : lack of nn.BatchNorm2d(ndf)
I'm learning DCGAN by translating [mxnet](http://preview.d2l.ai/d2l-en/master/chapter_generative-adversarial-networks/dcgan.html)  code to [pytorch](https://pytorch.org/tutorials/beginner/dcgan_faces_tutorial.html).

Snipaste_2020-08-16_20-15-13
I'm a newbie.
Then in my guess,  Discriminator has pretty much structures opposite to Generator.
So they must have same number layers.

Here is my code:
https://github.com/StevenJokes/d2l-en-read/blob/master/chapter-generative-adversarial-networks/dcgan_pt3.ipynb
My PR: https://github.com/pytorch/tutorials/pull/1125

### PR DESCRIPTION
**Pull request by StevenJokes**

### CODE REVIEW

**Comment by soumith** [comment]
hey, BatchNorm seems unnecessary for the first layer, as the inputs to the first Conv2d are already zero-mean and unit std, and the Conv2d is initialized (initially in a random draw) in a way that the output of the Conv2d roughly maintains zero-mean and unit std.

This is why we avoided putting batchnorm in the first layer in our original code / paper
